### PR TITLE
Bp 386 add filter bar on grid page

### PR DESCRIPTION
**Pull request by Munin33**

### CODE REVIEW

File: src/GridView/CardGrid.js [hunk 1]
```diff
@@ -28,6 +28,43 @@ class CardGrid extends ConnectionComponent {
...
+	doesContain(baseValue, subValue) {
+		return (baseValue.toString().toLowerCase().indexOf(subValue.toString().toLowerCase()) > -1);
+	}
+
+	getAttributeValues(childEntity){
+		let attributeArray = [];
+		childEntity.Attributes.forEach((attribute) => {
+			attributeArray.push(attribute.Value)
+		});
+		console.log(attributeArray);
```
Comment (hpibptlab) [inline src/GridView/CardGrid.js]: MAJOR Remove this logging statement. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS2228)

File: src/GridView/CardGrid.js [hunk 2]
```diff
@@ -28,6 +28,43 @@ class CardGrid extends ConnectionComponent {
...
+			return false;
+		}
+	}
+
+	doesContain(baseValue, subValue) {
+		return (baseValue.toString().toLowerCase().indexOf(subValue.toString().toLowerCase()) > -1);
+	}
+
+	getAttributeValues(childEntity){
+		let attributeArray = [];
```
Comment (hpibptlab) [inline src/GridView/CardGrid.js]: MINOR Make "attributeArray" "const". [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS3353)

File: src/GridView/CardGrid.js [hunk 3]
```diff
@@ -28,6 +28,43 @@ class CardGrid extends ConnectionComponent {
...
+	testFilter(childEntity, filter) {
+		if (!filter.value) {
+		 	return true;
+		 }
+		 const attributeValues = this.getAttributeValues(childEntity);
+
+		if (this.doesContain(childEntity.Name, filter.value)){
+			return true;
+		}
+		else if (this.doesContain(attributeValues, filter.value)){
```
Comment (hpibptlab) [inline src/GridView/CardGrid.js]: MAJOR Either merge this branch with the identical one on line "44" or change one of the implementations. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS1871)

File: src/GridView/GridView.js [hunk 4]
```diff
@@ -55,6 +77,9 @@ class GridView extends ConnectionComponent {
 			return connectionIncomplete;
 		}
 		const childEntityTypes = this.getChildEntityTypes(entity.TypeId, hierarchy);
+
+		let filterBar = this.loadFilterBar();
```
Comment (hpibptlab) [inline src/GridView/GridView.js]: MINOR Make "filterBar" "const". [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS3353)

**Comment by hpibptlab** [comment]
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: src/GridView/GridView.js [hunk 5]
```diff
@@ -3,18 +3,25 @@ import {Container} from "react-grid-system";
...
 import config from './../config/config.js';
 import Header from "../Header";
 
+
 class GridView extends ConnectionComponent {
 	constructor() {
 		super();
 		this.getChildEntityTypes = this.getChildEntityTypes.bind(this);
 		this.getEntityType = this.getEntityType.bind(this);
+		this.handleFilterChange = this.handleFilterChange.bind(this);
```
Comment (MaximilianV) [inline src/GridView/GridView.js]: I'm not sure, but I think it is a convention to but bindings at the bottom of the constructor.

File: src/GridView/GridView.js [hunk 6]
```diff
@@ -64,17 +89,18 @@ class GridView extends ConnectionComponent {
 						currentEntity={entity}
 						getEntityType={this.getEntityType}
 						getChildEntityTypes={this.getChildEntityTypes}/>
-					<SearchBar/>
+					{filterBar}
```
Comment (MaximilianV) [inline src/GridView/GridView.js]: If you have no condition for `filterBar` you can put the function call here directly instead of using a variable.

**Review by MaximilianV — CHANGES_REQUESTED** [review]
Looks good to me!

File: src/GridView/GridView.js (hunk 5 above)
Comment (tommartensen) [reply src/GridView/GridView.js]: https://medium.freecodecamp.com/react-binding-patterns-5-approaches-for-handling-this-92c651b5af56 -> Approach no. 4
1. super call
2. state definition
3. bindings

**Comment by frederike-ramin** [comment]
Replaced by #92.

**Pull request closed without merging**